src/v/kafka/server/group.cc

```diff
@@ -1353,6 +1359,52 @@ group::prepare_tx(cluster::prepare_group_tx_request r) {
     co_return cluster::prepare_group_tx_reply();
 }
 
+ss::future<cluster::abort_group_tx_reply>
+group::abort_tx(cluster::abort_group_tx_request r) {
+    // TODO: https://app.clubhouse.io/vectorized/story/2197
+    // (check for tx_seq to prevent old abort requests aborting
+    // new transactions in the same session)
+
+    auto tx = aborted_tx{.group_id = r.group_id, .tx_seq = r.tx_seq};
+
+    iobuf key;
+    kafka::response_writer w(key);
+    // TODO: https://app.clubhouse.io/vectorized/story/2200
+    // include producer_id+type into key to make it unique-ish
+    // to prevent being GCed by the compaction
+    w.write(aborted_tx_record_version());
+    storage::record_batch_builder builder(
+      cluster::group_abort_tx_batch_type, model::offset(0));
+    builder.set_producer_identity(r.pid.id, r.pid.epoch);
+    builder.set_control_type();
+    builder.add_raw_kw(
+      std::move(key),
+      reflection::to_iobuf(std::move(tx)),
+      std::vector<model::record_header>());
+    auto batch = std::move(builder).build();
+    auto reader = model::make_memory_record_batch_reader(std::move(batch));
+
+    auto e = co_await _partition->replicate(
+      std::move(reader),
+      raft::replicate_options(raft::consistency_level::quorum_ack));
+
+    if (!e) {
+        co_return make_abort_tx_reply(cluster::tx_errc::timeout);
+    }
+
+    auto vtx_it = _volatile_txs.find(r.pid);
+    if (vtx_it != _volatile_txs.end()) {
+        _volatile_txs.erase(vtx_it);
+    }
```

Comment:
I don't think you need to look up the key first. that is, `_volatile_txs.erase(r.pid)` is probably a no-op if it doesn't already exist. that avoids double look-up.